data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/util/ContentData.java

```diff
@@ -35,22 +41,42 @@
 
     public Map<String, Object> convertType(Map<String, Object> newProperties, List<String> contentModelListTypeFields, List<String> dateFields) {
         Map<String, Object> result = new HashMap();
-        for(String entry: contentModelListTypeFields){
-            if(newProperties.containsKey(entry)
+        for (String entry : contentModelListTypeFields) {
+            if (newProperties.containsKey(entry)
                     && newProperties.get(entry) instanceof String) {
                 String str = (String) newProperties.get(entry);
-                if(dateFields.contains(entry)) {
-                    try{
+                if (dateFields.contains(entry)) {
+                    try {
                         Date date = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss").parse(str);
-                        result.put(entry,date.getTime());
-                    }catch (ParseException ex){ ex.printStackTrace();}
-                }
-                else {
+                        result.put(entry, date.getTime());
+                    } catch (ParseException ex) {
+                        ex.printStackTrace();
+                    }
+                } else {
                     List<String> value = Arrays.asList(str);
                     result.put(entry, value);
                 }
             }
         }
         return result;
     }
+
+    public Object getMetadata(String apiUrl, RestUtil restUtil,String authKey, String fieldName) {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Authorization", authKey);
+        try {
+            okhttp3.Response httpResponse = restUtil.get(apiUrl, headers);
```

Comment:
We should unirest which is being used across all other projects instead of okhttp